relnote(116): support dirname on textarea and input elements (#28032)
mdn/content: files/en-us/web/html/element/textarea/index.md

  - : This Boolean attribute lets you specify that a form control should have input focus when the page loads. Only one form-associated element in a document can have this attribute specified.
- `cols`
  - : The visible width of the text control, in average character widths. If it is specified, it must be a positive integer. If it is not specified, the default value is `20`.
- `dirname`

  - : This attribute is used to set the text directionality of the element in a manner similar to the [`dirname`](/en-US/docs/Web/HTML/Element/input#dirname) attribute of the `<input>` element. Possible values include `ltr` for left-to-right and `rtl` for right-to-left. The default value is `ltr`.

>     The value of the attribute is sent in the format `{elementName}.dir={value}`, where `{elementName}` is the name of the `<input>` element and `{value}` may be `ltr` or `rtl`.

Author: oops :)
```suggestion
    The value of the attribute is sent in the format `{elementName}.dir={value}`, where `{elementName}` is the name of the `<textarea>` element and `{value}` may be `ltr` or `rtl`.
```

  - : This Boolean attribute lets you specify that a form control should have input focus when the page loads. Only one form-associated element in a document can have this attribute specified.
- `cols`
  - : The visible width of the text control, in average character widths. If it is specified, it must be a positive integer. If it is not specified, the default value is `20`.
- `dirname`

>   - : This attribute is used to set the text directionality of the element in a similar fashion to the [`<input>` element's `dirname` attribute](/en-US/docs/Web/HTML/Element/input#dirname).
>     The value is sent in the format `{elementName}.dir={value}` where `{value}` may be either `ltr` for left-to-right (default) or `rtl` for right-to-left.
>     The following element would send `info.dir=rtl` with form data if right-to-left text was entered into the field:

Reviewer: Suggesting to state the possible values and the default upfront
```suggestion
  - : This attribute is used to set the text directionality of the element in a manner similar to the [`dirname`](/en-US/docs/Web/HTML/Element/input#dirname) attribute of the `<input>` element. Possible values include `ltr` for left-to-right and `rtl` for right-to-left. The default value is `ltr`.
    The value of the attribute is sent in the format `{elementName}.dir={value}`, where `{elementName}` is the name of the `<input>` element and `{value}` may be `ltr` or `rtl`.
    The following element will send `info.dir=rtl` along with form data when text is entered into the form field:
```
"...if right-to-left text was entered into the form field": I believe when you type in the field, you still type left to right, but on submission the text is treated right to left? I tried a code sample with `rtl` but I was still typing left to right in the form field. Because only on submission of the attribute value will the form know how to handle the text?

Author: >I tried a code sample with rtl but I was still typing left to right in the form field marked as rtl.

🤔 - good call, I will get a demo together to try it out. Committing your prose improvements in the meantime

    ```html

>     <textarea cols="90" name="info" dirname="info.dir"></textarea>

Reviewer: ```suggestion
    <textarea cols="90" name="info" dirname="info.dir=rtl"></textarea>
```

Author: I think we're getting rid of the example, but the original is correct as it's working like this:

* `name="info"` identifies the textarea element by name
* `dirname="info.dir"` when the form is submitted, the server can use this (`info.dir={direction}`) to tell what text direction the element with `info` name has. The `=rtl` bit depends on what the user submits, not the author.

Reviewer: The lead-in to the example said "The following element would send `info.dir=rtl` with form..." and so I thought it needs to be the same in the example.

From what I gathered after reading this https://www.w3.org/International/questions/qa-html-dir#reportingdirection (linked from the Firefox nightly release notes page), I guess `dir` sets the direction of text you're typing and `dirname` is used later to treat the text after form submission?

> The directional information can then be used to apply the correct direction to the text when it is displayed on another page.

>     The following element would send `info.dir=rtl` with form data if right-to-left text was entered into the field:
> 
>     ```html
>     <textarea cols="90" name="info" dirname="info.dir"></textarea>
>     ```

Reviewer 2: ```suggestion
```
Examples should be included in the example section, not attribute list. Also, this example adds importance to this attribute when none is needed.

Reviewer: Sorry, @estelle, this time I missed seeing your self-assignment on this PR, plus also didn't assign it to myself when I started reviewing.

Author: >Examples should be included in the example section, not attribute list. Also, this example adds importance to this attribute when none is needed.

Very good, going to get rid of it - it feels out of place indeed, note that we're doing the same thing on the [input `dirname` attr](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#dirname) and `name` which was why I added it. Maybe this should be moved out in a follow-up, what do you both reckon?

Author: Removed in ec543ff4639f434252016929a952f0ad7796a84c

Reviewer: > Maybe this should be moved out in a follow-up, what do you both reckon?

Yeah, I agree, would be good in a follow-up PR. Probably a separate page for the `dirname` attribute might help with this. As a side note, the `<input>` element page does not have an H2 Examples section.
